rotkehlchen/chain/optimism/decoding/decoder.py

```diff
@@ -43,6 +47,10 @@ def __init__(
                 address_is_exchange_fn=self._address_is_exchange,
             ),
         )
+        self.dbevmtx = DBOptimismTx(database)
```

Comment:
That is because the superclass `EVMTransactionDecoder` already initializes this here: https://github.com/rotki/rotki/blob/63d49f2501114376d5a9f727b083b280fa092f4e/rotkehlchen/chain/evm/decoding/decoder.py#L148

But for the optimism decoder you need a different tx class. You are overriding it here. It's not totally wrong, but it would be cleaner to parametrize the constructor to initialize the different class.